### ISSUE TITLE
PostgreSQL ON CONFLICT .. WHERE .. DO NOTHING generates not valid SQL

### ISSUE THREAD

**Issue by runeflobakk** [jOOQ/jOOQ#13660]
This issue is almost identical to the fixed #11732, only that it seems jOOQ still qualifies the columns if using `.doNothing()` instead of `doUpdate()`. I get the same error message as described in the mentioned issue: **ERROR: invalid reference to FROM-clause entry for table "X". There is an entry for table "X", but it cannot be referenced from this part of the query.**, and I see that the generated SQL fully qualifies the columns in the WHERE-clause for ON CONFLICT. If I change to `doUpdate()`, and include some "dummy" `.set(..)`, the generated query is valid.

### Expected behavior

```java
...
.onConflict(TABLE.COLUMN)
.where(TABLE.COLUMN.eq("value")).doNothing()
```
should generate SQL:
```sql
on conflict ("column") where "column" = ? do nothing 
```

### Actual behavior

This SQL is generated:
```sql
on conflict ("column") where "db"."table"."column" = ? do nothing 
```

### Versions

- jOOQ: 3.16.6
- Java: Eclipse Adoptium 17.0.3
- Database (include vendor): PostgreSQL 11.15
- OS: MacOS
- JDBC Driver (include name if inofficial driver): postgresql 42.3.6

**Comment by lukaseder** [comment]
Thanks a lot for your report. Indeed, I can reproduce this. Will fix right away.

**Comment by runeflobakk** [comment]
Yes, it has indeed been fixed in PostgreSQL, but maybe not backported I guess?

**Comment by lukaseder** [comment]
There: https://twitter.com/pg_xocolatl/status/1382099254341144583

**Comment by lukaseder** [comment]
Yeah, well, it doesn't seem like the most significant bug

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.17.0, 3.16.7 (https://github.com/jOOQ/jOOQ/issues/13661), and 3.15.11 (https://github.com/jOOQ/jOOQ/issues/13662)

**Comment by runeflobakk** [comment]
Wow, lightning speed! ⚡ 
Thank you!
